src/drv_nxtvr/nxtvr.c

```diff
@@ -0,0 +1,240 @@
+    {
+        LOGE("NXTVR: REPORT size has to be at least 8!");
+        return;
+    };
+
+    uint8_t reportID = (uint8_t)buffer[0];
+    if (reportID == HID_REPORT_QUAT_ID)
+    {
+        if (size != HID_REPORT_QUAT_SIZE)
+        {
+            LOGE("NXTVR: Invalid data size %d for reportID 42.", size);
+            return;
+        };
+
+        quatf quater;
+        //copy remaining 16 bytes into float union
+        memcpy((void *)&quater, (void *)&buffer[1], 16);
+
+        priv->base.rotation = quater;
+        LOGD("NXT: %f %f %f %f", quater.x, quater.y, quater.z, quater.w);
+    }
+    else
+    {
+        LOGW("RELATIV: Unknown reportID %d.", reportID);
+    }
+}
+
+static void update_device(ohmd_device *device)
+{
+    nxt_priv *priv = (nxt_priv *)device;
+
+    int size = 0;
+    unsigned char buffer[FEATURE_BUFFER_SIZE];
+
+    if (!priv->handle)
+    {
+        LOGE("NXTVR Device handle missing in update_device fnc!");
+        return;
+    }
+
+    while (true)
+    {
+        int size = hid_read(priv->handle, (void *)&buffer, FEATURE_BUFFER_SIZE);
+        if (size < 0)
+        {
+            LOGW("NXTVR: update_device - error reading device.");
+            return;
+        }
+        else if (size == 0)
+        {
+            return;
+        }
+
+        handle_nxtvr_hmd_sensor_msg(priv, buffer, size);
+    }
+}
+
+static int getf(ohmd_device *device, ohmd_float_value type, float *out)
+{
+    nxt_priv *priv = (nxt_priv *)device;
+
+    switch (type)
+    {
+    case OHMD_ROTATION_QUAT:
+        *(quatf *)out = priv->base.rotation;
+        break;
+    case OHMD_POSITION_VECTOR:
+        out[0] = out[1] = out[2] = 0;
+        break;
+    case OHMD_DISTORTION_K:
+        memset(out, 0, sizeof(float) * 6);
+        break;
+    case OHMD_CONTROLS_STATE:
+        out[0] = .1f;
+        out[1] = 1.0f;
+        break;
+    default:
+        ohmd_set_error(priv->base.ctx, "invalid type given to getf (%ud)", type);
+        return OHMD_S_INVALID_PARAMETER;
+        break;
+    }
+    return OHMD_S_OK;
+}
+
+static void close_device(ohmd_device *device)
+{
+    nxt_priv *priv = (nxt_priv *)device;
+    LOGD("closing nxtvr device");
+    if (!priv->handle)
+    {
+        hid_close(priv->handle);
+        priv->handle = NULL;
+    }
+
+    free(device);
+}
+
+static ohmd_device *open_device(ohmd_driver *driver, ohmd_device_desc *desc)
+{
+    nxt_priv *priv = ohmd_alloc(driver->ctx, sizeof(nxt_priv));
+
+    if (!priv)
+        return NULL;
+
+    priv->handle = hid_open_path(desc->path);
+    if (!priv->handle)
+    {
+        LOGD("Clould not open directory '" DEVICE_NAME "'.");
+        goto err;
+    }
+
+    if (hid_set_nonblocking(priv->handle, 1) == -1)
+    {
+        LOGD("Clould not set nonblocking on '" DEVICE_NAME "'.");
+        goto err_handle;
+    }
+
+    // Set default device properties
+    ohmd_set_default_device_properties(&priv->base.properties);
+
+    // Set device properties (imitates the rift values)
```

Comment:
Due to this data being a rift copy, is this driver even ready?
If this is still in development i would recommend maintaining this in your own repository until the headset is finalized, before adding this into Master.